src/stratcon_jlog_streamer.c

```diff
@@ -98,6 +109,83 @@ struct jlog_stream_stats {
 };
 mtev_hash_table noit_stats_iep; /* this is persistent */
 mtev_hash_table noit_stats_durable; /* this is persistent */
+mtev_hash_table noit_consul_service;
+
+/* consul actuation */
+static void consul_start(const char *cn, const struct ipport *tgt) {
+  if(mtev_consul_service_alloc_available()) {
+    if(strlen(tgt->target)) { /* we only start if we have an address */
+      mtev_hash_table *tags = calloc(1, sizeof(*tags));
+      mtev_hash_init(tags);
+      mtev_hash_dict_store(tags, "cn", cn);
+      mtev_consul_service *service =
+        mtev_consul_service_alloc("noit", cn, tgt->target, tgt->port,
+            tags, true, NULL, false);
+      mtev_consul_service_check_https(service, "/capa.json", "GET", cn, true, 10, NULL);
+      mtev_consul_register(service);
+    }
+  }
+}
+static void consul_stop(const char *cn) {
+  if(mtev_consul_service_registry_available()) {
+    service_register *sr = mtev_consul_service_registry(cn);
+    if(sr) {
+      struct ipport tgt = { .port = 0, .target = "unknown" };
+      void *vtgt = NULL;
+      pthread_mutex_lock(&noit_ip_by_cn_lock);
+      if(mtev_hash_retrieve(&noit_ip_by_cn, strdup(cn), strlen(cn),
+                            &vtgt)) {
+        tgt = *(struct ipport *)vtgt;
+      }
+      pthread_mutex_unlock(&noit_ip_by_cn_lock);
+      consul_start(cn, &tgt);
```

Comment:
I find it confusing that a function called ``consul_stop`` is calling ``consul_start``

Reply:
That's fair. I'll add a comment.

Reply:
Actually, it's likely we want to add a deregister and split the behavior.  It will require a bit more work.